rust/lance-encoding/src/buffer.rs

```diff
@@ -398,30 +300,26 @@ impl LanceBuffer {
     /// Unlike `slice_with_length`, this method allows for slicing at a bit level but always
     /// requires a copy of the data (unless offset is byte-aligned)
     ///
-    /// This method also converts to a borrowed buffer for convenience, but that could be optimized
-    /// away in the future if needed.
-    ///
     /// This method performs the bit slice using the Arrow convention of *bitwise* little-endian
     ///
     /// This means, given the bit buffer 0bABCDEFGH_HIJKLMNOP and the slice starting at bit 3 and
     /// with length 8, the result will be 0bNOPABCDE
     pub fn bit_slice_le_with_length(&mut self, offset: usize, length: usize) -> Self {
-        let Self::Borrowed(borrowed) = self.borrow_and_clone() else {
-            unreachable!()
-        };
         // Use this and remove backport once we upgrade to arrow-rs 54
```

Comment:
I think we're on arrow-rs 54 now.  We can probably remove this backport (can be done in follow-up)